pkg/cmd/server/origin/master_config.go

```diff
@@ -213,6 +219,26 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
 	return config, nil
 }
 
+func newControllerPlug(options configapi.MasterConfig, client *etcdclient.Client) (plug.Plug, func()) {
+	switch {
+	case options.ControllerLeaseTTL > 0:
+		id := kapi.SimpleNameGenerator.GenerateName("master-")
+		leaser := leaderlease.NewEtcd(
+			client,
+			path.Join(options.EtcdStorageConfig.OpenShiftStoragePrefix, "leases/controllers"),
```

Comment:
You can have a key that is a dir (I think).  So leases/controllers and
leases/controllers/foo should be able to coexist.  I *think* we're going to
move to an API version one so this should be less of a problem.

On Fri, Aug 14, 2015 at 4:09 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/origin/master_config.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37114438>:
>
> > @@ -213,6 +219,26 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
> >  	return config, nil
> >  }
> >
> > +func newControllerPlug(options configapi.MasterConfig, client *etcdclient.Client) (plug.Plug, func()) {
> > +	switch {
> > +	case options.ControllerLeaseTTL > 0:
> > +		id := kapi.SimpleNameGenerator.GenerateName("master-")
> > +		leaser := leaderlease.NewEtcd(
> > +			client,
> > +			path.Join(options.EtcdStorageConfig.OpenShiftStoragePrefix, "leases/controllers"),
>
> leases/allcontrollers in case we later want leases/controllers/foo?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37114438>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift